test/frontend/helpers/setup.js

```
@@ -0,0 +1,283 @@
+  document.body.innerHTML = MIN_HTML;
+  _installStubs();
+
+  // Reset state via the in-scope helper function (window.eval can't reach
+  // let-vars from a prior eval's lexical environment).
+  global._testSetState({}, [], null);
+
+  // Restore any functions that were mocked in previous tests
+  for (const [name, fn] of Object.entries(_originals)) {
+    const tmpKey = `__orig_${name}`;
+    global[tmpKey] = fn;
+    window.eval(`${name} = ${tmpKey};`); // eslint-disable-line no-eval
```

Biome (2.4.6)

[error] eval() exposes to security risks and performance issues.

(lint/security/noGlobalEval)